cheevos/cheevos.c

```diff
@@ -2984,35 +2987,29 @@ bool rcheevos_load(const void *data)
       coro->data       = NULL;
       coro->path       = strdup(info->path);
 
-      /* if we're looking at an m3u file, get the first disc from the playlist */
-      if (string_is_equal_noncase(path_get_extension(coro->path), "m3u"))
+      /* Check whether this is an m3u file */
+      if (m3u_file_is_m3u(coro->path))
       {
-         intfstream_t* m3u_stream = intfstream_open_file(coro->path, RETRO_VFS_FILE_ACCESS_READ, RETRO_VFS_FILE_ACCESS_HINT_NONE);
-         if (m3u_stream)
-         {
-            char m3u_contents[1024];
-            char disc_path[PATH_MAX_LENGTH];
-            char* tmp;
-            int64_t num_read = intfstream_read(
-                  m3u_stream, m3u_contents, sizeof(m3u_contents) - 1);
+         m3u_file_t *m3u_file = m3u_file_init(coro->path, M3U_FILE_SIZE);
```

Comment:
Would it be reasonable to only request 1 item here? It looks like the entire file is still going to be processed anyway.